### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by vermasavita**

### CODE REVIEW

**Review by JowelTisso — APPROVED** [review]
* Clean code
* Looks good

File: src/reducer/auth-reducer.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    case "CHECK_USER":
+      return {
+        ...state,
+        user: action.payload.user,
+        token: action.payload.token,
+      };
+    case "LOGOUT":
+      return { ...state, user: null, payload: null };
+    default:
+      return state;
```
Comment (Kevin-Solomon) [inline src/reducer/auth-reducer.js:26]: You can also throw an Error object here instead of returning the state.
But for now, this also works.

**Review by mohammed-sas — APPROVED** [review]
As mentioned by Kevin, fix those issues
Other than that your code is clean
Auth context has been coded good

File: src/context/auth-context.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { createContext, useContext, useReducer } from "react";
+import { useEffect } from "react";
```
[the remark this answers is not in the document]
Comment (vermasavita) [reply src/context/auth-context.js:2]: Done

File: src/pages/authentication/SignUp.jsx [hunk 3]
```diff
@@ -1,70 +1,140 @@
...
+    setUser({ ...user, [id]: value });
+  };
+
+  const checkAllIptFields = () => {
+    return user.email !== "", user.password !== "", user.confirmPassword !== "";
+  };
+
+  // const checkEmail = () => {
+  //   validator.isEmail(user.email) ? true : false;
+  // };
```
[the remark this answers is not in the document]
Comment (vermasavita) [reply src/pages/authentication/SignUp.jsx:28]: I'll use this code later to validate email address

File: src/reducer/auth-reducer.js (hunk 1 above)
Comment (vermasavita) [reply src/reducer/auth-reducer.js:26]: Error object as in? 
Can you explain it a bit more?

Comment (Kevin-Solomon) [reply src/reducer/auth-reducer.js:26]: You can read more on Error objects.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error

**Pull request merged**